src/Features/CSharp/Portable/CodeRefactorings/ConvertToRecord/CSharpConvertToRecordRefactoringProvider.cs

```diff
@@ -217,24 +220,33 @@ private static SyntaxList<AttributeListSyntax> GetModifiedAttributeListsForPrope
         /// modifies constructors and some method modifiers to fall in line with record requirements (e.g. this() initializer)
         /// </summary>
         /// <param name="typeDeclaration">Original type declaration</param>
+        /// <param name="type"></param>
         /// <param name="semanticModel">Semantic model</param>
-        /// <param name="propertiesToMove">Properties we decided to move</param>
-        /// <param name="cancellationToken">Cancellation token</param>
-        /// 
+        /// <param name="propertiesToMove">Properties we decided to move, may return in a different order</param>
+        /// <param name="defaults">properties in order for positional record with associated defaults</param>
+        /// <param name="cancellationToken">cancellationToken</param>
         /// <returns>The list of members from the original type, modified and trimmed for a positional record type usage</returns>
         private static ImmutableArray<MemberDeclarationSyntax> GetModifiedMembersForPositionalRecord(
             TypeDeclarationSyntax typeDeclaration,
             INamedTypeSymbol type,
             SemanticModel semanticModel,
             ImmutableArray<PositionalParameterInfo> propertiesToMove,
+            out ImmutableArray<(PropertyDeclarationSyntax property, EqualsValueClauseSyntax? @default)> defaults,
```

Comment:
Can we avoid the out in favor of this having a richer return type? A record, maybe? :smile:

Reply:
I'll just move the contents of this method into the one that calls it (`ConvertToPositionalRecordAsync`) since they virtually have the same parameter list and both methods are very long already. I think the separate method abstraction made sense when it was self-contained, but now that there are multiple values that get returned/set I think it's better to just integrate it all together.